### PR TITLE
Fixes quickswap letting you keep clothing traits

### LINKED ISSUE #50798: Using the "swap equipped item" button, it's possible to retain HUDs from eyewear that you've removed
doublehud

### LINKED ISSUE #52788: Quick-Swapping glasses can cure nearsightedness & stack huds
## Reproduction:
Using the quickswap feature, you are capable of swapping between clothes without needing to take them off. While this is overall a good feature, there are slight bugs which this can entail. The below photos show the process for recreating said issue.
Starting state 1
Starting state 2
Starting state 3
Firstly, you can see that we start with a character who is nearsighted, and requires glasses in order to not have their vision obscured. This is normal.
Issue state 1
Issue state 2
However, if we are to quick-swap from one eye-slot item to another, we suddenly find that our nearsightedness is magically cured, even if we take off the new pair of goggles/glasses. This is _not_ normal.
Issue state 3
Finally, if we are to quick-swap between different HUD items, we see that we suddenly have gained the ability to see every aspect of a person's identity & current state (the diagnostic HUD is similar, however because nothing nearby actually has nanites or is a borg, it doesn't show up properly.)
I am going to assume that the reason nearsightedness is affected is due to the glasses effect being related to it being dropped from the eye slot, which it does not recognize has occurred if quick-swap is used.

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
 Unequipping clothing via quickswap now properly calls dropped(), so you lose any traits you get for wearing them. This fixes being able to use the quickswap hotkey to keep traits and other properties like krav maga, sec/med/diagnostic hud's, and whatever else without actually wearing the clothing.

Fixes: #52788
Fixes: #50798

## Why It's Good For The Game
Honestly for my money, a quickswap hotkey is lame and overly gamey, which is funny because the kind of player who would be into that is exactly the kind of player I'd suspect of quietly abusing a bug like this bug for the last several months, but I digress.

## Changelog
:cl: Ryll/Shaps
fix: Quickswapping clothing now properly unequips the swapped-out clothing, removing any traits they granted while worn
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
This bug was yuck, I'm genuinely surprised it went under the radar for so long.

**Comment by ATH1909** [comment]
> This bug was yuck, I'm genuinely surprised it went under the radar for so long.

It wasn't under the radar; people reported it multiple times, but it took until now for a coder to actually bother to fix it.

I'm gonna miss this; being able to see who's bad AND who's hurt at the same time (without being a CMO, being a silicon, or needing special eyes from robotics) was really nice while it lasted.

**Comment by Jared-Fogle** [comment]
rip <3

**Comment by Pugie** [comment]
Equipment swapping is fuckin' YUCK. Having to drop your items when swapping clothes is what we call quality gameplay.

**Comment by Ryll-Ryll** [comment]
> Equipment swapping is fuckin' YUCK. Having to drop your items when swapping clothes is what we call quality gameplay.

imagine having to make yourself vulnerable for a very brief moment of time in a roleplaying game while changing your clothing/armor, a task that in real life usually requires your full attention and leaves you vulnerable to being ambushed or attacked while doing so. i sure am glad that our gameflow isn't interrupted by any of those small inconveniences that mirror real life and force you to pay attention to your surroundings for even 5 seconds, cause those 5 seconds are way beyond my attention span

anyway, neatened up species equipping code and fixed another issue with quickswapping letting you drop abstract and etc stuff

**Comment by optimumtact** [comment]
> I'm gonna miss this; being able to see who's bad AND who's hurt at the same time (without being a CMO, being a silicon, or needing special eyes from robotics) was really nice while it lasted.
 
> Honestly for my money, a quickswap hotkey is lame and overly gamey, which is funny because the kind of player who would be into that is exactly the kind of player I'd suspect of quietly abusing a bug like this bug for the last several months, but I digress.

nailed it lmao

**Comment by necromanceranne** [comment]
Did ATHATH both report the bug and also admit to actually continuing to exploit it? Jesus fucking christ, where is the shame.

Also I honestly hate this feature and genuinely think it shouldn't exist for basically the same reasons ryll stated.

**Comment by optimumtact** [comment]
does it not take time to do? I thought the tradeoff was you had to stop and let it process?

**Comment by Jared-Fogle** [comment]
It does. It is legitimately faster to do it manually.

**Comment by Ryll-Ryll** [comment]
Not for things like armor or clothing where you can have other stuff sitting in the attached slots that would normally drop to the floor

**Review by Rohesie — CHANGES_REQUESTED** [review]
This is failing the `outfit_sanity` unit test.

This whole system is evil. If you want to rework it let me know and I'll give you pointers. Else, oh well, it is what it is.

**Comment by ShizCalev** [comment]
should put_in_hand() call dropped if it's location if the item's location is a mob instead?

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -923,50 +923,37 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
 
+	var/obj/item/replaced_item = H.get_item_by_slot(slot)
+	if(replaced_item && !swap)
+		return FALSE
+
+	if(replaced_item && ((replaced_item.item_flags & ABSTRACT) || (replaced_item.item_flags & DROPDEL) || HAS_TRAIT(I, TRAIT_NODROP)))
+		return FALSE
+
+	// these slots aren't necessarily replacing anything, since they're just storage slots. We do their own swap checks below
+	if(!(slot in list(ITEM_SLOT_SUITSTORE, ITEM_SLOT_BACKPACK)) && !(I.slot_flags & slot) && !(I.w_class <= WEIGHT_CLASS_SMALL && (slot == ITEM_SLOT_RPOCKET || slot == ITEM_SLOT_LPOCKET))) // we can fit anything small or below in pockets
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species.dm:934]: This is a mess, split this into two checks at least and describe what they do.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (AnturK) [reply code/modules/mob/living/carbon/human/species.dm:934]: @Ryll-Ryll please, this is one line change.

**Pull request merged**